src/Microsoft.AspNetCore.Server.Kestrel/Internal/Http/SocketInput.cs

```diff
@@ -189,10 +203,16 @@ public MemoryPoolIterator ConsumingStart()
                 {
                     if (!consumed.IsDefault)
                     {
+                        var lengthConsumed = new MemoryPoolIterator(_head).GetLength(consumed);
+
                         returnStart = _head;
                         returnEnd = consumed.Block;
                         _head = consumed.Block;
                         _head.Start = consumed.Index;
+
+                        // Must call Subtract() after bytes have been freed, to avoid producer starting too early and growing
+                        // buffer beyond max length.
```

Comment:
So I think it's fine leaving the code as it is. I was just trying to suggest a clearer comment. I think the comment you have now is fine.